### ISSUE TITLE
New property in ModifiedOrder

### ISSUE THREAD

**Issue by aclfe** [checkstyle/checkstyle#19513]
I have read check documentation: https://checkstyle.org/checks/modifier/modifierorder.html#ModifierOrder
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

different style guide implies modified order should be in such a manner: 

```
Modifiers should go in the following order
Access modifier (public / private / protected)
abstract
static
final
transient
volatile
default
synchronized
native
strictfp
``` 

As of now, ModifierOrder does not follow this order. Checkstyle's check `ModifierOrder` can extend a property to allow user to change ModifiedOrder in their own style guide rather than only default 

  **Is your feature request related to a problem?**
  
  ModifierOrder only enforces the JLS-suggested modifier order. Different style guides (e.g. OpenJDK) require a different order, but there is no way to configure the order - users either accept the JLS default or disable the check
  entirely.

  **Describe the solution you'd like**

  Add a new property order to ModifierOrder that accepts an explicit ordered list of modifier keywords. The default remains the existing JLS order (backward compatible). Users who want a different style configure it in XML.

  Config using default (JLS) order - no change needed:
```xml
  <module name="Checker">
    <module name="TreeWalker">
      <module name="ModifierOrder"/>
    </module>
  </module>
```
  Config using OpenJDK style (user writes the order explicitly):
```xml
  <module name="Checker">
    <module name="TreeWalker">
      <module name="ModifierOrder">
        <property name="modifierOrder"
            value="public, protected, private, abstract, static, final,
                   transient, volatile, default, synchronized, native, strictfp"/>
      </module>
    </module>
  </module>
```

  Incorrect order (OpenJDK config above):
```java
  public class Test {
      private final static String S = "x"; // violation, static must come before final
  }
```
  Correct order:
```java
  public class Test {
      private static final String S = "x"; // ok
  }
```

**Comment by aclfe** [comment]
I'm on it

**Comment by romani** [comment]
@aclfe , in what style guide different order is required ?

We need prove that somebody need this, as new property will be API commitment that we will have to support for years, and it complicates implementation. It should be a good reason.

**Comment by aclfe** [comment]
@romani OpenJDK Style Guide

https://checkstyle.org/styleguides/openjdk-java-style-v6/openjdk-styleguide.html#s3.5-modifiers

The issue was made before the parent issue https://github.com/checkstyle/checkstyle/issues/19604 so overlapping

**Comment by romani** [comment]
@aclfe , pelase update issue description it as new feature request.

example https://github.com/checkstyle/checkstyle/issues/19813 , attention to comment where we show how config will looks like.

**Comment by aclfe** [comment]
Updated the description, I'll also update the code

**Comment by Aman-Baliyan** [comment]
@romani can i take this issue there is no response on this pr -> https://github.com/checkstyle/checkstyle/pull/20888

**Comment by Aman-Baliyan** [comment]
on it

**Comment by romani** [comment]
fix is merged
